### ISSUE TITLE
Line marks getting distorted on export to PowerBI

### ISSUE THREAD

**Issue by stopyoukid** [Microsoft/charticulator-extensions#3]
It should look like this:

![image](https://user-images.githubusercontent.com/2513737/44681120-8eefe180-a9f4-11e8-8949-e3d047ed1566.png)

It looks like this in PowerBI:
![image](https://user-images.githubusercontent.com/2513737/44681143-a0d18480-a9f4-11e8-86d0-d62921f5c94b.png)

Here is the PBIX, rename from .zip to .pbiviz
[charticulator (76).zip](https://github.com/Microsoft/charticulator-extensions/files/2325332/charticulator.76.zip)

This *seems* to work fine if I draw the line mark DIRECTLY on the data axis between the min/max values, and it snaps vertically to the data axis.  Which leads me to believe that the line that I drew in this issue was slightly diagonal, and the powerbi rendering is just blowing that out of proportion (but it looks fine in charticulator)

**Comment by donghaoren** [comment]
Yes, in general the exported visual may have more layout issues than the original chart, because we have to compute layout from scratch. Make sure both ends of the line are snapped properly before exporting. Otherwise, charticulator will have a hard time doing the layout.

To mitigate these kind of issues, we can add a mechanism to store the average current values as defaults, so the exported visual can at least have something to start with. But, to make a perfect template, the user still has to snap things properly...

**Comment by stopyoukid** [comment]
Forgive me for my naiveté, but it seems like this could be solved by just storing the angle of the line, and using that in the layout calculations.  But, another option: update the "line" mark type to snap to specific angles (0, 15, 30, 45, 60 ...) so it is much harder to screw it up, and by default when you snap the both points to a data value (i.e. min, max), charticulator will automatically make it a vertical line.  Would that even help? Do users actually need to snap to a vertical line for layout to work correctly and the angles don't really matter?

After looking at the example in the gallery (https://charticulator.com/gallery/boston_weather.html), I see now that you are using the vertical line for snapping.  It's not clear from the video that doing that would be a requirement.  Especially since, when I did it without using the vertical line, the chart looked just fine in Charticulator, it only looked skewed in the export.

I guess the crux of the issue is that exported visuals should look identical (given the same dataset) to the one that you created in charticulator.

**Comment by donghaoren** [comment]
Angle will only makes sense in a Cartesian coordinates. In polar coordinates x, y actually means angle (polar angle) and radius (distance to the center), but `atan2(radius, angle)` won't make sense mathematically, as radius and angle don't share the same unit.

I think the best way is still to extract a good default value (x, y, dx, dy for lines, if they are not mapped to data or snapped) from the original chart, and use that as a starting point for layout calculation in the visual.

**Comment by stopyoukid** [comment]
Awesome, glad we have some way to alleviate it.

Yes, I understand that it doesn't make sense in polar coordinates.  I guess I just meant the slope of the line, so a 45 degree angle would just be 1 and 0 would be 0.

How this would apply to the stacked angular layout: you start at the angle around the circle where the data point belongs, use that as the center point for the line mark, and then draw a line using cartesian coordinates rotated by the angle you are at on the circle and use the outer radius - the inner radius to scale the line appropriately based on its bound data.

See the image below for what currently happens, even though I'd expect an uncurved line (using stacked angular, stacked radial totally makes sense to curve):
![screen shot 2018-09-05 at 10 55 13 am](https://user-images.githubusercontent.com/2513737/45111660-5d64cd80-b0fa-11e8-8ec7-a33d853c60ce.png)

That being said, I'm happy to go with whatever you think is best.  

So, I did notice something that was weird, that may be contributing to the skewing.

Below is before I applied the Polar scaffold, my line is bound to min & max (just not snapped to a vertical line), but it is well within the glyph's grid.

![image](https://user-images.githubusercontent.com/2513737/45114009-bafc1880-b100-11e8-882d-14bc759e8dbd.png)

After I apply the polar scaffold (and select stacked angular), the mark moves outside of the glyph's grid and to the left, and so it might be contributing to the skewing.

![image](https://user-images.githubusercontent.com/2513737/45114049-cbac8e80-b100-11e8-9bbf-97362625cbc5.png)

Now if I then move the line inside of the glyph's grid, it seems to export correctly (or at least the skewing is much smaller and I cannot see it)
